README.md

```diff
@@ -12,17 +12,35 @@ The Azure [validator](https://github.com/spectrocloud-labs/validator) plugin ens
 
 The Azure validator plugin reconciles `AzureValidator` custom resources to perform the following validations against your Azure environment:
 
-TODO
+1. Compare the Azure RBAC permissions associated with a [security principal](https://learn.microsoft.com/en-us/azure/role-based-access-control/overview#security-principal) against an expected permission set.
 
 Each `AzureValidator` CR is (re)-processed every two minutes to continuously ensure that your Azure environment matches the expected state.
 
 See the [samples](https://github.com/spectrocloud-labs/validator-plugin-azure/tree/main/config/samples) directory for example `AzureValidator` configurations.
 
-(TODO: actual samples)
+## Authn & Authz
 
-## Supported Service Quotas by Azure Service
+Authentication details for the Azure validator controller are provided within each `AzureValidator` custom resource. Azure authentication can be configured either implicitly or explicitly. All supported options are detailed below:
 
-TODO
+* Implicit (`AzureValidator.auth.implicit == true`)
+  * [Environment variables](https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication?tabs=bash#-option-1-define-environment-variables)
```

Comment:
Thanks, that makes sense. Azure actually has another form of authentication that I didn't add here yet called "user-assigned" managed identities. The idea is that you either explicitly opt in to this auth method when you use the SDK (we don't - we instead use "Default" and let the SDK cascade through methods) or you use the "Default" cascading and specify env vars in a way where it cascades past the "three env vars" method, then past workload identity, and then into user-assigned managed identity (because you'd only specify CLIENT_ID env var, no more and no less).

Any idea how we could tackle that with our setup? Or should we pass on that for now?

Reply:
Sounds like that should already be possible by specifying only CLIENT_ID in the secret.